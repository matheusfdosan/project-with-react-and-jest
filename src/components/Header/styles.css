#header {
  width: 100%;
  padding: 1rem;
  border-bottom: 0.1rem solid #3f3f3f49;
  text-align: center;
}

#header-container > h1 {
  font-family: "Playfair Display", serif;
  font-size: 3rem;
  font-weight: 500;
  color: #3f3f3f;
  user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
}
