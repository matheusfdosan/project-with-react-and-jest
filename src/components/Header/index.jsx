import React from "react"
import "./styles.css"

export default function Header() {
  return (
    <div id="header">
      <div id="header-container">
        <h1>UrbanVault</h1>
      </div>
    </div>
  )
}
