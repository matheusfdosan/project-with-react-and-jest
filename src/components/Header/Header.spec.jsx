import "@testing-library/jest-dom"
const { render, screen } = require("@testing-library/react")
import React from "react"
import Header from "."

describe("<Header />", () => {
  it("should render header whith store name", () => {
    render(<Header />)

    const heading = screen.getByRole("heading", { level: 1 })

    expect(heading).toBeInTheDocument()
    expect(heading).toHaveTextContent("UrbanVault")
  })

  it("should match snapshot", () => {
    const { container } = render(<Header />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
