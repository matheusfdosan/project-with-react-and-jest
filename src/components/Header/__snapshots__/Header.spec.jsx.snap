// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot 1`] = `
<div
  id="header"
>
  <div
    id="header-container"
  >
    <h1>
      UrbanVault
    </h1>
  </div>
</div>
`;
